src/pages/POS/POS.vue

```diff
@@ -934,17 +956,21 @@ export default defineComponent({
           await this.applyPricingRule();
           await this.sinvDoc.runFormulas();
           if (isInventoryItem) {
+            const completeItem = (await this.fyo.doc.getDoc(
+              ModelNameEnum.Item,
+              item.name
+            )) as Item;
             await validateQty(
               this.sinvDoc as SalesInvoice,
-              item as Item,
+              completeItem,
```

Comment:
```suggestion
              item as POSItem,
```

Replace completeItem with `item as POSItem`